app/models/ability.rb

```diff
@@ -60,6 +60,14 @@ def initialize(user)
         can? :teach, sub.course
       end
 
+      can :download, Submission do |sub|
+        !sub.course.hide_submission_result? && can?(:read, sub)
```

Comment:
Why the downloading of a zip of the exercise depends on whether the course allows the user to view results?
(I assume it is the download for the zip)